File: roslib/java/ros/Publisher.java
package com.roslib.ros;

import com.roslib.tinyros_msgs.TopicInfo;

public class Publisher<MsgT extends Msg> {
    public java.lang.String topic_;
    public MsgT msg_;
    public long id_;
    public NodeHandleBase nh_;
    public boolean negotiated_;

    public long endpoint_;

    public Publisher(java.lang.String topic_name, MsgT msg) {
        this.endpoint_ = TopicInfo.ID_PUBLISHER;
        this.topic_ = topic_name;
        this.msg_ = msg;
		this.negotiated_ = false;
    }

    public int publish(MsgT msg) {
        if (this.nh_ != null) {
            return this.nh_.publish(this.id_, msg);
        }
        return 0;
    }

    public long getEndpointType() {
        return this.endpoint_;
    }

    public boolean negotiated() {
        return this.negotiated_;
    }
}

File: roslib/javascript/tinyros/Publisher.js
(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f();}else if(typeof define==="function"&&define.amd){define([],f);}else{var g;if(typeof window!=="undefined"){g=window;}else if(typeof global!=="undefined"){g=global;}else if(typeof self!=="undefined"){g=self;}else{g=this;}if(typeof g.tinyros==="undefined"){g.tinyros={};}g.tinyros.Publisher=f();}})(function(){var define,module,exports;'use strict';
const _CURRENT_PATH_ = document.currentScript.src.substring(0, document.currentScript.src.lastIndexOf("/"));
const _ROOT_PATH_ = _CURRENT_PATH_.substring(0, _CURRENT_PATH_.lastIndexOf("/"));
document.write("<script language=javascript src='"+_ROOT_PATH_+"/tinyros_msgs/TopicInfo.js'></script>");

function Publisher(topic_name, msg_type) {
    this.id = -1;
    this.negotiated = false;
    this.nh = null;
    this.msg = msg_type;
    this.topic = topic_name;
    this.endpoint = tinyros_msgs.TopicInfo().ID_PUBLISHER;
};

Publisher.prototype.publish = function(msg) {
    if (this.nh !== null) {
        return this.nh.publish(this.id, msg);
    } else {
        return -1;
    }
};
  
Publisher.prototype.getMsgType = function() {
    return this.msg().getType();
};

Publisher.prototype.getMsgMD5 = function() {
    return this.msg().getMD5();
};

Publisher.prototype.getEndpointType = function() {
    return this.endpoint;
};

Publisher.prototype.negotiated = function() {
    return this.negotiated;
};

return function(_1,_2) { return new Publisher(_1,_2); };

});





